core/src/main/java/io/grpc/Metadata.java

```diff
@@ -235,7 +305,25 @@ public boolean containsKey(Key<?> key) {
   public <T> void put(Key<T> key, T value) {
     Preconditions.checkNotNull(key, "key");
     Preconditions.checkNotNull(value, "value");
-    storeAdd(key.name, new MetadataEntry(key, value));
+    if (len() == 0 || len() == cap()) {
+      int desiredCap = len() + (len() >>> 1) + 4;
```

Comment:
I wanted to avoid the max call, to avoid conditions.  Really the ideal growth strategy is from multiplying by [phi](https://crntaylor.wordpress.com/2011/07/15/optimal-memory-reallocation-and-the-golden-ratio/), (or adding the two previous sizes together), but 1.5 is a very easy to calculate approximation.  The plus 4 is to make sure we don't ever get an odd number when expanding a second time.